AppDB/datastore_server.py

```diff
@@ -3653,10 +3655,30 @@ def __extract_entities_from_composite_indexes(self, query, index_result):
       if definition.ancestor() == 1:
         ancestor = tokens.pop(0)[:-1]
       distinct_str = ""
+      value_index = 0
       for def_prop in definition.property_list():
-        value = tokens.pop(0) 
+        # If the value contained the separator, try to recover the value.
+        if len(tokens[:-1]) != len(definition.property_list()):
```

Comment:
Question: could that ever be that definition.propety.list() is bigger than the possible number of values in tokens (ie wrong query or something like that)?